src/lib/features/feature-search/feature-search-store.ts

```diff
@@ -326,6 +327,43 @@ class FeatureSearchStore implements IFeatureSearchStore {
                 'ranked_features.feature_name',
                 'lifecycle.stage_feature',
             );
+        if (this.flagResolver.isEnabled('flagsOverviewSearch')) {
+            finalQuery
+                .leftJoin(
+                    this.db('change_request_events AS cre')
+                        .join(
+                            'change_requests AS cr',
+                            'cre.change_request_id',
+                            'cr.id',
+                        )
+                        .select('cre.feature')
+                        .select(
+                            this.db.raw(
+                                'array_agg(distinct cre.change_request_id) AS change_request_ids',
+                            ),
+                        )
+                        .select('cr.environment')
+                        .groupBy('cre.feature', 'cr.environment')
+                        .whereNotIn('cr.state', [
```

Comment:
the tradeoff here is that the query needs to be aware of the states to exclude. Since we don't need to add new states very often I think it's ok.